data/resources/ui/schedule_rule_edit_dialog.ui

```diff
@@ -192,6 +192,74 @@
                   </object>
                 </child>
 
+                <child>
+                  <object class="AdwPreferencesGroup">
+                    <property name="title" translatable="yes">Pre and Post Snapshot scripts</property>
+                    <property name="description" translatable="yes">Scripts to run before and after the snapshot,&#10;Scripts get the source and target paths passed as arguments,&#10;If the Pre Snapshot script is provided and it fails then the backup would not be performed,&#10;Please note that the scripts are run as root and you may cause damage to your system if you are not careful.</property>
+
+                    <child>
+                      <object class="AdwActionRow">
+                        <property name="title" translatable="yes">Pre Snapshot</property>
```

Comment:
I agree, my goal with this PR was to experiment and try out ways to move and manage backups on external location,
and that needed all the 4 script hooks that I added.
I am trying to model my setup similar to Time Machine on macOS and its backup and cleanup scheme for snapshots and external backups.
And I have realized that a backup tool like that does not fit Butter's workflow exactly,
I think making the pre / post prune fields experimental and not showing them in UI right now would make sense right now,
I think we should keep the pre snapshot as it might be useful in some usecases and pre / post hooks around an action is a common pattern in general but also in other backup tools.